File: backend/src/main/java/org/fiesc/felipe/backend/modules/model/entity/Pessoa.java
package org.fiesc.felipe.backend.modules.model.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "pessoas")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pessoa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idPessoa;

    @NotBlank
    private String nome;

    private LocalDate nascimento;

    @Column(unique = true)
    private String cpf;

    @Email
    private String email;

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "id_endereco")
    private Endereco endereco;

    @Column(name = "situacao_integracao", nullable = false)
    private String situacaoIntegracao;
}

File: frontend/src/types/Pessoa.ts
import type { SituacaoIntegracaoKey } from "../config/integrationStatus";
import type { Endereco } from "./Endereco";

export type Pessoa = {
    id: string;
    nome: string;
    nascimento?: string;
    cpf?: string;
    email?: string;
    endereco: Endereco;
    status: SituacaoIntegracaoKey;
  };